Some machine and kernel cleanup/optimizations

MaxKellermann opened the pull request.

kedder commented on recipes-bsp/u-boot/files/0001-Environment-Openvario-mainline.patch:
```diff
@@ -80,8 +80,8 @@ index 9819d99..76e7a83 100644
-+			"setenv bootargs console=ttyS0,115200 root=/dev/mmcblk0p2 rootwait panic=10 fbcon=rotate:${rotation_kernel};" \
-+		"else setenv bootargs console=ttyS0,115200 root=/dev/mmcblk0p2 rootwait panic=10;" \
++			"setenv bootargs console=ttyS0,115200 root=/dev/mmcblk0p2 rootwait panic=10 lima.job_hang_limit=10 fbcon=rotate:${rotation_kernel};" \
++		"else setenv bootargs console=ttyS0,115200 root=/dev/mmcblk0p2 rootwait panic=10 lima.job_hang_limit=10;" \
```
I'd really wish we would preserve the reason for this `lima.job_hang_limit=10` somewhere. Not sure what's the best way to do this, but it is getting harder and harder to trace it back to #71.

MaxKellermann replied:
Agree, that's important, I'll move the code comment over into this u-boot patch.
Unfortunately, u-boot scripts are already unwieldy; then you have a u-boot script inside a C string literal, and if this weren't complicated enough, this u-boot script inside a C string literal is then inside a patch file. This needs to improve. There should be a plain text file with the script.

kedder reviewed (approved):
Looks good to me otherwise.

MaxKellermann commented:
I added an explanation to the u-boot patch description (to be moved somewhere else when the u-boot script becomes a separate file), and I added another bunch of kernel configuration changes.

MaxKellermann commented:
@linuxianer99?

linuxianer99 commented:
> @linuxianer99?

Sorry ... i did a review yesterday, but missed to commit the review ... sorry ... my fault ...

MaxKellermann commented on recipes-kernel/linux/linux-mainline/cpufreq.cfg:
```diff
@@ -7,4 +7,14 @@ CONFIG_CPU_FREQ_DEFAULT_GOV_SCHEDUTIL=y
 CONFIG_CPU_FREQ_GOV_SCHEDUTIL=y
 CONFIG_CPUFREQ_DT=y
-# CONFIG_CPU_IDLE is not set
+
```
No. It just puts the CPU or other devices into power-saving states when they are not used. This does not mean suspend or hibernate.

MaxKellermann replied:
If you mean `CONFIG_SUSPEND` - that indeed is suspend to RAM. This is not needed, nor is `SUSPEND_FREEZER` or `PM_AUTOSLEEP` needed or used for anything right now. I could remove those options now, but I thought they might be worth experimenting with. For example, if the glider is on the ground, you could have an option to suspend it until you actually take off, saving some battery meanwhile.

MaxKellermann commented on conf/machine/openvario-43-rgb.conf:
```diff
@@ -27,7 +27,7 @@ PREFERRED_PROVIDER_virtual/bootloader ?= "u-boot"
 KERNEL_IMAGETYPE ?= "uImage"
 
 IMAGE_CLASSES += "sdcard_image-openvario"
-IMAGE_FSTYPES += "ext3 tar.gz openvario-sdimg"
```
But this tarball cannot be used for building anything. It does not contain any development files.

linuxianer99 replied:
Hmm ... you are right.

MaxKellermann commented on conf/machine/ov-cubieboard2.inc:
```diff
@@ -0,0 +1,19 @@
+KERNEL_DEVICETREE = "openvario.dtb"
+#KERNEL_DEVICETREE = "sun7i-a20-cubieboard2.dtb"
+UBOOT_MACHINE = "openvario_defconfig"
+SUNXI_FEX_FILE = "sys_config/a20/cubieboard2.fex"
```
This line was copied from meta-sunxi/cubieboard2.conf, and my PR *removes* it; instead of copying stuff from other repositories, meta-openvario should inherit (and extend) those files. Copying stuff is rarely ever a good idea; it causes big problems when updating to new Yocto versions.
